### DISCUSSION TITLE
Unable to restore mindstorms inventor hub to lego firmware after installing pybricks

### DISCUSSION THREAD

**Discussion by pruehr** [pybricks/support#1778] [Q&A]
I have successfully installed pybricks on my lego mindstorms inventor hub.  However when I try to restore to lego firmware, I get an error when trying to update the lego firmware with Lego app.  I have tried just plugging the hub into usb and using pybricks fixit but my light never turns pink.  Am using mac m2 pro computer.

**Comment by dlech** [comment]
You have to restore an older version of LEGO firmware first using the Pybricks Code app.

Did you follow the procedure pictured below?

<img width="790" alt="image" src="https://github.com/user-attachments/assets/f7ed225b-3483-4556-bc21-66abcc0b79ac">

**Reply by pruehr** [reply]
![Screenshot 2024-08-25 at 3 34 11 PM](https://github.com/user-attachments/assets/e610aa33-b95c-482c-b743-9086eae917fc)
YES, I keep getting message, something went wrong, to drop device from bluetooth and restart.  I have tried this many times and always get same result.  I am able to load pybricks sucessfully and run code but cannot restore to lego firmware.

**Reply by dlech** [reply]
When you turn on the hub, what color is the power button?

**Reply by laurensvalk** [reply]
You have to follow the steps David referred to _in Pybricks_ to install the original firmware again. Click on **restore official LEGO Firmware** as shown below, and then follow the indicated steps.

![image](https://github.com/user-attachments/assets/5c8edb82-e33e-469d-a89c-7ab8b1f88ad5)

Since you were able to install successfully, restoring using these steps will also work. It uses the exact same technique, with the same button combination to activate update mode.

**Comment by pruehr** [comment]
blue

From: David Lechner ***@***.***>
Date: Sunday, August 25, 2024 at 3:56 PM
To: pybricks/support ***@***.***>
Cc: pruehr ***@***.***>, Author ***@***.***>
Subject: Re: [pybricks/support] Unable to restore mindstorms inventor hub to lego firmware after installing pybricks (Discussion #1778)

When you turn on the hub, what color is the power button?

**Comment by pruehr** [comment]
I did follow those steps and pybricks says it finished the update. When I connect to the Mindstorms app, it starts to update, then I get the message that something went wrong.

From: laurensvalk ***@***.***>
Date: Sunday, August 25, 2024 at 4:21 PM
To: pybricks/support ***@***.***>
Cc: pruehr ***@***.***>, Author ***@***.***>
Subject: Re: [pybricks/support] Unable to restore mindstorms inventor hub to lego firmware after installing pybricks (Discussion #1778)

You have to follow the steps David referred to in Pybricks to install the original firmware again. Click on restore official LEGO Firmware as shown below, and then follow the indicated steps.

image.png (view on web)<https://github.com/user-attachments/assets/5c8edb82-e33e-469d-a89c-7ab8b1f88ad5>

Since you were able to install successfully, restoring using these steps will also work. It uses the exact same technique, with the same button combination to activate update mode.

**Reply by dlech** [reply]
Sometimes the LEGO app will say that the update failed even when it really actually succeeded. If the light on the hub is no longer blue, then the LEGO firmware has been restored.

**Comment by pruehr** [comment]
In my case the light is still blue and the hub will not shut off, I have to pull the battery to shut it off.

From: David Lechner ***@***.***>
Date: Sunday, August 25, 2024 at 4:53 PM
To: pybricks/support ***@***.***>
Cc: pruehr ***@***.***>, Author ***@***.***>
Subject: Re: [pybricks/support] Unable to restore mindstorms inventor hub to lego firmware after installing pybricks (Discussion #1778)

Sometimes the LEGO app will say that the update failed even when it really actually succeeded. If the light on the hub is no longer blue, then the LEGO firmware has been restored.

**Reply by dlech** [reply]
Does the light blink Blue or is it on steady all of the time? 

Would it be possible to share a video of what you are doing for the restore procedure pictured below (both the computer screen and the LEGO hub) so we can see where it is going wrong?


<img width="662" alt="image" src="https://github.com/user-attachments/assets/0524c39d-d32c-4703-9dd0-bb323fa44c29">

**Comment by pruehr** [comment]
Sure I can do that probably late tomorrow or Tuesday.  Thanks so much for
all of your help.  Am really over my head with all of this!  Also, this
afternoon I built your ball balancer for inventor and the motors just ran
on high speed like the gyro was not working, don’t know if this is related.

**Comment by pruehr** [comment]
I made the video but it is too big to email, will try to figure a solution this afternoon.

From: David Lechner ***@***.***>
Date: Sunday, August 25, 2024 at 5:39 PM
To: pybricks/support ***@***.***>
Cc: pruehr ***@***.***>, Author ***@***.***>
Subject: Re: [pybricks/support] Unable to restore mindstorms inventor hub to lego firmware after installing pybricks (Discussion #1778)

Does the light blink Blue or is it on steady all of the time?

Would it be possible to share a video of what you are doing for the restore procedure pictured below (both the computer screen and the LEGO hub) so we can see where it is going wrong?
image.png (view on web)<https://github.com/user-attachments/assets/0524c39d-d32c-4703-9dd0-bb323fa44c29>
